src/main/kotlin/dev/optimistic/decentdiscordbridge/DecentDiscordBridge.kt

```diff
@@ -1,56 +1,19 @@
-import com.google.gson.Gson
-import com.google.gson.reflect.TypeToken
-import dev.minn.jda.ktx.events.listener
-import dev.minn.jda.ktx.jdabuilder.light
-import dev.optimistic.decentdiscordbridge.Configuration.AvatarConfiguration.TemplateType.*
-import dev.optimistic.decentdiscordbridge.avatars.AbstractAvatarUrlGenerator
-import dev.optimistic.decentdiscordbridge.avatars.impl.DashedUuidArgumentUrlGenerator
-import dev.optimistic.decentdiscordbridge.avatars.impl.UndashedUuidArgumentUrlGenerator
-import dev.optimistic.decentdiscordbridge.avatars.impl.UsernameArgumentUrlGenerator
-import dev.optimistic.decentdiscordbridge.ducks.CachedAvatarUrlDuck
-import dev.optimistic.decentdiscordbridge.filter.FilterRenderer
-import dev.optimistic.decentdiscordbridge.filter.impl.AppliedFilterRenderer
-import dev.optimistic.decentdiscordbridge.filter.impl.NoOpFilterRenderer
-import dev.optimistic.decentdiscordbridge.mention.AbstractMentionResolver
-import dev.optimistic.decentdiscordbridge.mention.impl.DisabledMentionResolver
-import dev.optimistic.decentdiscordbridge.mention.impl.EnabledMentionResolver
-import dev.optimistic.decentdiscordbridge.message.DiscordMessageToMinecraftRenderer
-import dev.optimistic.decentdiscordbridge.util.StringExtensions.escapeDiscordSpecial
+import dev.optimistic.decentdiscordbridge.discord.AbstractBridge
+import dev.optimistic.decentdiscordbridge.discord.impl.DisabledBridge
+import dev.optimistic.decentdiscordbridge.discord.impl.EnabledBridge
 import me.lucko.configurate.toml.TOMLConfigurationLoader
-import net.dv8tion.jda.api.JDA
-import net.dv8tion.jda.api.entities.Guild
-import net.dv8tion.jda.api.entities.UserSnowflake
-import net.dv8tion.jda.api.events.message.MessageReceivedEvent
-import net.dv8tion.jda.api.requests.GatewayIntent
-import net.dv8tion.jda.api.requests.RestAction
-import net.dv8tion.jda.api.utils.MemberCachePolicy
 import net.fabricmc.loader.api.FabricLoader
-import net.minecraft.network.message.SignedMessage
 import net.minecraft.server.PlayerManager
-import net.minecraft.server.network.ServerPlayerEntity
-import net.minecraft.text.Text
 import org.slf4j.LoggerFactory
 import org.spongepowered.configurate.kotlin.extensions.get
 import org.spongepowered.configurate.kotlin.objectMapperFactory
 import java.nio.file.Files
-import java.nio.file.Path
 
-class DecentDiscordBridge(private val playerManager: PlayerManager) {
-    val logger = LoggerFactory.getLogger("decent-discord-bridge")
-    private val filter: FilterRenderer
-    val urlGenerator: AbstractAvatarUrlGenerator
-    private val allowedMentions: AllowedMentions
-    private val webhook: JDAWebhookClient
-    private val jda: JDA
-    private val mentionResolver: AbstractMentionResolver
-    val seenUsers: MutableSet<Long>
-    private val seenUsersPath: Path
+class DecentDiscordBridge(playerManager: PlayerManager) {
+    private val logger = LoggerFactory.getLogger("decent-discord-bridge")
+    val impl: AbstractBridge
```

Comment:
This should replace the bridge property in the companion object, such that the property type definition is changed to that of the abstract class